receiver/datadogreceiver/metrics_translator.go

```diff
@@ -13,12 +15,104 @@ import (
 
 type MetricsTranslator struct {
 	sync.RWMutex
-	buildInfo component.BuildInfo
-	lastTs    map[identity.Stream]pcommon.Timestamp
+	buildInfo  component.BuildInfo
+	lastTs     map[identity.Stream]pcommon.Timestamp
+	stringPool *StringPool
 }
 
 func newMetricsTranslator() *MetricsTranslator {
 	return &MetricsTranslator{
-		lastTs: make(map[identity.Stream]pcommon.Timestamp),
+		lastTs:     make(map[identity.Stream]pcommon.Timestamp),
+		stringPool: newStringPool(),
 	}
 }
+
+func (mt *MetricsTranslator) streamHasTimestamp(stream identity.Stream) (pcommon.Timestamp, bool) {
+	mt.RLock()
+	defer mt.RUnlock()
+	ts, ok := mt.lastTs[stream]
+	return ts, ok
+}
+
+func (mt *MetricsTranslator) updateLastTsForStream(stream identity.Stream, ts pcommon.Timestamp) {
+	mt.Lock()
+	defer mt.Unlock()
+	mt.lastTs[stream] = ts
+}
+
+const (
+	TypeGauge string = "gauge"
+	TypeRate  string = "rate"
+	TypeCount string = "count"
+)
+
+type SeriesList struct {
+	Series []datadogV1.Series `json:"series"`
+}
+
+func translateMetricsV1(series SeriesList, mt *MetricsTranslator) pmetric.Metrics {
```

Comment:
only used in context of `MetricsTranslator`, can this be a method? e.g. `mt.TranslateV1`